### ISSUE TITLE
Update to Scala 2.12

### ISSUE THREAD

**Issue by tkw1536** [UniFormal/MMT#200]
As discussed with @florian-rabe we should investigate if updating our scala version to 2.12 will break any of our code. For this we should in particular have a look at deprecation warnings. If this is not the case, we should be able to use Scala 2.12 instead of 2.11.

**Comment by tkw1536** [comment]
This should be fixed after #3.

**Comment by florian-rabe** [comment]
To make two major groups of feature warnings disappear, we should compile with the options:
"-language:implicitConversions -language:postfixOps"

**Comment by florian-rabe** [comment]
I went over deprecation warnings in the important projects (I didn't check all of them though.).

I had to change code in only a few places to respond to Scala deprecations.
Essentially all warnings that remain are our own deprecations.

**Comment by florian-rabe** [comment]
I tried compiling with Scala 2.12.

Not all projects like 2.12 yet, but I've removed all compilation errors in the most important ones.

**Comment by florian-rabe** [comment]
On my own machine, I'm now using

* Java 1.8
* ScalaIDE 4.7.0
* Scala 2.12.3

without major problems.

The sbt workflow still uses Scala 2.11, and some projects do not compile with 2.12 yet.

Still, the problem seems easier than expected. So I've moved it one milestone earlier.
If individual projects turn out to be too hard to upgrade, we may have to push this issue back again.

**Comment by tkw1536** [comment]
@florian-rabe 
> To make two major groups of feature warnings disappear, we should compile with the options:
> "-language:implicitConversions -language:postfixOps"

What speaks against adding

```Scala
import scala.language.postfixOps
import scala.language.implicitConversions
```

to the appropriate files?

**Comment by florian-rabe** [comment]
Those imports are just useless boilerplate that would have to be added to many files.

**Comment by florian-rabe** [comment]
I've officially moved to Scala 2.12.3 and sbt 1.0.4 [*]
I think I'm still targeting Java 8. But that discussion can continue in issue #263.

'sbt deploy' still works (after I deprecated one unused project and removed it from build.sbt).
So I've pushed new deploy/lib/*.jar and build.sbt to devel that work with Scala 2.12.
@Jazzpirate Some code in archives may still have to be adapted.

We can close this unless there are further issues.

[*]  We can't move to sbt 1.1.0 because it's buggy on Windows. We have to wait for sbt 1.1.1, where the bug is supposedly fixed.
